### ISSUE TITLE
Unit test a rule that is disabled by default?

### ISSUE THREAD

**Issue by pdb-stripe** [pinterest/ktlint#2281]
I have been looking at ways to use `ktlint-test` and my own mechanism for spinning up a `KtlintRuleEngine`, but there does not appear to be a simple way to unit test a custom rule that is disabled in the default `.editorconfig`.

## Expected Behavior
I'd like to be able to create a custom rule, disable it by default, then unit test that it works (when it is not disabled). For example, let's say I have a custom rule that prevents me from importing "com.github.foo", and I have it configured for `ktlint:my-rules:import-foo`. I set this as disabled in my default editorconfig:

`ktlint_my-rules_import-foo = disabled`

Then I write a unit test against it. It might be something like this:

````kotlin
@Test
fun testFoo() {
    val latch = CountDownLatch(1)
    var result: LintError? = null
    KtLintRuleEngine(setOf(RuleProvider { FooImportRule() })).lint(Code.fromSnippet("import com.github.foo")) {
        result = it
        latch.countDown()
    }
    latch.await(1, TimeUnit.SECONDS)
    assertNotNull(result) // Fails here
}
````

The expectation is that the test should pass

## Actual behavior

The only way it passes is if I remove the line I configured in `.editorconfig`, such that the rule is not disabled.

## Additional information
* Current version of ktlint: 0.50.0

**Comment by paul-dingemans** [comment]
When using the `ktlint-test` module, you can write your unit tests using the [ktlintAssertThat](https://github.com/pinterest/ktlint/blob/a679dd4f2e84288e98d6d06a42091bcf0b0087a8/ktlint-test/src/main/kotlin/com/pinterest/ktlint/test/KtLintAssertThat.kt#L4) functionality. This is used by almost all unit tests in ktlint.

The missing piece, might be how to disable a rule. For most properties this is simple, but for disabling a rule it is a bit more complex. It can be done as follows:
```
internal class SomeRuleTest {
    private val someRuleAssertThat = assertThatRule { SomeRule() }

    @Test
    fun `Some test`() {
        val code =
            """
            // some code with violations that are detected by SomeRule if it is executed
            """.trimIndent()
        someRuleAssertThat(code)
            .withEditorConfigOverride(SomeRule().ruleId.createRuleExecutionEditorConfigProperty() to RuleExecution.enabled)
            .hasViolations(
                // check your violations
            ).isFormattedAs(
               // formatted code
           )
    }
}
```

But in my opionion this should not be tested for each rule that you implement. It is logic that is tested by ktlint with some rule. See https://github.com/pinterest/ktlint/blob/a679dd4f2e84288e98d6d06a42091bcf0b0087a8/ktlint-rule-engine/src/test/kotlin/com/pinterest/ktlint/rule/engine/internal/rulefilter/RuleExecutionRuleFilterTest.kt#L4
